src/cache/cache.service.ts

```diff
@@ -44,34 +50,89 @@ export class CacheService<T extends unknown> {
     return join(this.cacheDir, networkDir);
   }
 
-  private async getCacheFilePath(): Promise<string> {
+  private getCacheFileName(batchIndex: number): string {
+    return `${batchIndex}.${this.cacheFile}`;
+  }
+
+  private async getCacheFilePaths(): Promise<string[]> {
     const dir = await this.getCacheDirPath();
-    return join(dir, this.cacheFile);
+    const existCacheFiles: string[] = [];
+
+    let batchIndex = 0;
+
+    do {
+      try {
+        const filePath = join(dir, this.getCacheFileName(batchIndex));
+        const fileStat = await stat(filePath);
+
+        if (!fileStat.isFile()) return existCacheFiles;
+        existCacheFiles.push(filePath);
+
+        batchIndex++;
+      } catch (error) {
+        return existCacheFiles;
+      }
+    } while (true);
   }
 
-  private async getCacheFromFile(): Promise<T> {
+  private async getCacheFromFiles(): Promise<T> {
     try {
-      const filePath = await this.getCacheFilePath();
-      const content = await readFile(filePath);
-      return JSON.parse(String(content));
+      const filePaths = await this.getCacheFilePaths();
+
+      let headers = this.cacheDefaultValue.headers as H;
+      let data = [] as D[];
+
+      await Promise.all(
+        filePaths.map(async (filePath) => {
+          const content = await readFile(filePath);
+          const parsed = JSON.parse(String(content));
+
+          if (
+            JSON.stringify(headers) !== JSON.stringify(parsed.headers) &&
+            headers !== this.cacheDefaultValue.headers
+          ) {
+            throw new Error('Headers are not equal');
+          }
+
+          headers = parsed.headers;
+          data = data.concat(parsed.data);
+        }),
+      );
+
+      return { headers, data } as T;
     } catch (error) {
       return this.cacheDefaultValue;
     }
   }
 
-  private async saveCacheToFile(): Promise<void> {
+  private async saveCacheToFiles(): Promise<void> {
+    if (!this.cache) throw new Error('Cache is not set');
+
+    const { headers, data } = this.cache;
+
     const dirPath = await this.getCacheDirPath();
-    const filePath = await this.getCacheFilePath();
     await mkdir(dirPath, { recursive: true });
 
-    return await writeFile(filePath, JSON.stringify(this.cache));
+    let batchIndex = 0;
+
+    do {
+      const from = batchIndex * this.cacheBatchSize;
+      const to = (batchIndex + 1) * this.cacheBatchSize;
+      const batchedData = data.slice(from, to);
+
+      if (batchedData.length === 0) return;
+
+      const filePath = join(dirPath, this.getCacheFileName(batchIndex));
+      await writeFile(filePath, JSON.stringify({ headers, data: batchedData }));
+
+      batchIndex++;
+    } while (true);
```

Comment:
I would suggest using a stricter loop, without the "while true".
And we need to consider the data integrity scenario. I suggest specifying the number of chunks in the headers.

Reply:
replaced with a for loop